parlai/core/loader.py

```diff
@@ -0,0 +1,325 @@
+        module_name = "%s.tasks.%s.agents" % (repo, task)
+
+    task_module = importlib.import_module(module_name)
+
+    return task_module
+
+
+def load_teacher_module(taskname: str):
+    """
+    Get the module of the teacher agent specified by `--task`.
+
+    Can be formatted in several different ways:
+
+    * full: ``-t parlai.tasks.babi.agents:DefaultTeacher``
+    * shorthand: ``-t babi``, which will check
+        ``parlai.tasks.babi.agents:DefaultTeacher``
+    * shorthand specific: ``-t babi:task10k``, which will check
+        ``parlai.tasks.babi.agents:Task10kTeacher``
+
+    The base path to search when using shorthand formats can be changed from
+    "parlai" to "parlai_internal" by prepending "internal:" to the path, e.g.
+    "internal:babi".
+
+    Options can be sent to the teacher by adding an additional colon,
+    for example ``-t babi:task10k:1`` directs the babi Task10kTeacher to use
+    task number 1.
+
+    :param taskname: path to task class in one of the above formats.
+
+    :return:
+        teacher module
+    """
+    task_module = load_task_module(taskname)
+    task_path_list, repo = _get_task_path_and_repo(taskname)
+
+    if len(task_path_list) > 1 and '=' not in task_path_list[1]:
+        task_path_list[1] = task_path_list[1][0].upper() + task_path_list[1][1:]
+        teacher = task_path_list[1]
+        if '.' not in task_path_list[0] and 'Teacher' not in teacher:
+            # Reformat from underscore to CamelCase and append "Teacher" to
+            # class name by default if a complete path is not given.
+            words = teacher.split('_')
+            teacher_name = ''
+            for w in words:
+                teacher_name += w[0].upper() + w[1:]
+            teacher = teacher_name + "Teacher"
+    else:
+        teacher = "DefaultTeacher"
+
+    teacher_class = getattr(task_module, teacher)
+    return teacher_class
+
+
+##############################################################
+### WORLD LOADER
+##############################################################
+def _get_default_world(default_world=None, num_agents=None):
+    """
+    Get default world if a world is not already specified by the task.
+
+    If a default world is provided, return this. Otherwise, return
+    DialogPartnerWorld if there are 2 agents and MultiAgentDialogWorld if
+    there are more.
+
+    :param default_world:
+        default world to return
+    :param num_agents:
+        number of agents in the environment
+    """
+    if default_world is not None:
+        world_class = default_world
+    elif num_agents is not None:
+        import parlai.core.worlds as core_worlds
+
+        world_name = (
+            "DialogPartnerWorld" if num_agents == 2 else "MultiAgentDialogWorld"
+        )
+        world_class = getattr(core_worlds, world_name)
+    else:
+        return None
+
+    return world_class
+
+
+def load_world_module(
+    taskname: str,
+    interactive_task: bool,
```

Comment:
another nit: maybe we could `interactive_task` be an optional argument, as in the test below it is a little confusing that this is just `True` or `False`

Reply:
i think it should just default to `False`, yes